src/connectors/WalletConnectConnector/index.ts

```diff
@@ -0,0 +1,146 @@
+import { UserRejectedRequestError } from '@pangolindex/web3-react-injected-connector';
+import WalletProvider, { EthereumProvider } from '@walletconnect/ethereum-provider';
+import { AbstractConnector } from '@web3-react/abstract-connector';
+import { ConnectorUpdate } from '@web3-react/types';
+
+export interface WalletConnectConnectorArguments {
+  rpcMap: { [chainId: number]: string };
+  projectId: string;
+  metadata: {
+    name: string;
+    description: string;
+    url: string;
+    icons: string[];
+    verifyUrl?: string;
+  };
+}
+
+export class WalletConnectConnector extends AbstractConnector {
+  public provider?: WalletProvider;
+  private config: WalletConnectConnectorArguments;
+
+  constructor(config: WalletConnectConnectorArguments) {
+    const supportedChainIds = Object.keys(config.rpcMap).map((k) => Number(k));
+    super({ supportedChainIds: supportedChainIds });
+    this.config = config;
+    this.handleChainChanged = this.handleChainChanged.bind(this);
+    this.handleAccountsChanged = this.handleAccountsChanged.bind(this);
+    this.handleDisconnect = this.handleDisconnect.bind(this);
+
+    if ((this.config.projectId, length > 0)) {
+      this.init();
+    }
+  }
+
+  private async init() {
+    const supportedChainIds = this.getSupportedChainIds();
+
+    const rpcMap = Object.entries(this.config.rpcMap).reduce((acc, [chainId, rpc]) => {
+      acc[`eip155:${chainId}`] = rpc;
+      return acc;
+    }, {} as { [chainId: string]: string });
+
+    this.provider = await EthereumProvider.init({
+      projectId: this.config.projectId,
+      chains: [43114],
```

Comment:
do we need to add other chains here ?

Reply:
Maybe, I tested with all chains but doesn't work, I think this need in first element the default chain and the rest of chains in numeric order

https://github.com/Uniswap/web3-react/blob/fc84e8b64b094b758f0c29a33977a47a349065bb/packages/walletconnect-v2/src/index.ts#L95C8-L95C8

Reply:
I didn't get you

Reply:
on `@web3-react/walletconnectv2` the chain key the first element of array is the default key, the rest of elements is another supported chains in numeric order for example, [43114, 1,10,16,20,...]